app/updatemanager.go

```diff
@@ -386,6 +409,81 @@ func (c *ControlMapPool) ClearExpired() {
 		},
 	)
 	c.Pool = newPool
+
+	c.saveToStore()
+}
+
+type controlMapPoolDbFormat struct {
+	Active  []*UpdateControlMap `json:"active"`
+	Expired []*UpdateControlMap `json:"expired"`
+}
+
+// Must only be called from functions that have already locked the mutex.
+func (c *ControlMapPool) saveToStore() {
+	if c.store == nil {
+		// Saving not enabled. This should not happen outside tests.
+		return
+	}
+
+	active, expired := c.getAll()
+
+	toSave := controlMapPoolDbFormat{
+		// Intentionally not using label assignment here. We want to
+		// know if we're missing any fields.
+		active,
+		expired,
+	}
+
+	data, err := json.Marshal(&toSave)
+	if err != nil {
+		log.Errorf("Could not marshal Update Control Maps to JSON: %s", err.Error())
+		// There isn't much we can do if it fails.
+		return
+	}
+
+	log.Debugf("Saving Update Control Maps to disk: %q", string(data))
+
+	err = c.store.WriteAll(datastore.UpdateControlMaps, data)
+	if err != nil {
+		log.Errorf("Could not save Update Control Maps to database: %s", err.Error())
+		// There isn't much we can do if it fails.
+		return
+	}
+}
+
+// `timeout` is the timeout that should be given to expired maps that we load
+// from disk.
+func (c *ControlMapPool) LoadFromStore(timeout int) {
```

Comment:
I think I might have made a mistake with the `stamp` interface. Perhaps it would have been better if it took a `time.Duration` parameter all along?

Reply:
Instead of `int`, you mean? Maybe, but I think it's fine as it is. I already made the mistake that `time.Duration(1)` is one nanosecond, not one second, which is what we use as the unit in the config file. So using the int, which forces conversion, is a nice reminder to multiply it with `time.Second`.

Reply:
Makes sense. Let's keep it